src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -518,8 +706,22 @@ export class InputTimePicker
   connectedCallback() {
     connectLocalized(this);
 
-    if (this.value) {
-      this.setValue({ value: isValidTime(this.value) ? this.value : undefined, origin: "loading" });
+    let { effectiveLocale } = this;
+    if (effectiveLocale === "en-US") {
+      effectiveLocale = "en";
+    }
+    if (effectiveLocale === "pt-PT") {
+      effectiveLocale = "pt";
+    }
+    if (effectiveLocale === "no") {
+      effectiveLocale = "nb";
+    }
+    this.effectiveLocale = effectiveLocale;
+
+    if (isValidTime(this.value)) {
+      this.setValueDirectly(this.value);
```

Comment:
Not critical, but I think we should have one method for setting the value internally to avoid diverging paths since both have common logic.

Reply:
Yeah, that's what the previous `setValue` pattern was doing, which required some kind of parameter that could tell the function whether to emit a change event or not.  We can explore that pattern since it applies to lots of other value-setting form components.